test/framework/src/main/java/org/elasticsearch/test/AbstractSerializingTestCase.java

```diff
@@ -60,15 +61,9 @@ public void testFromXContent() throws IOException {
     public void testToString() throws IOException {
         for (int runs = 0; runs < NUMBER_OF_TEST_RUNS; runs++) {
             T testInstance = createTestInstance();
-            if (testInstance instanceof ToXContentToBytes) {
-                XContentType xContentType = XContentType.JSON;
-                String toString = testInstance.toString();
-                assertParsedInstance(xContentType, new BytesArray(toString), testInstance);
-                for (Map.Entry<String, T> alternateVersion : getAlternateVersions().entrySet()) {
-                    String instanceAsString = alternateVersion.getKey();
-                    assertParsedInstance(XContentType.JSON, new BytesArray(instanceAsString), alternateVersion.getValue());
-                }
-            }
+            XContentType xContentType = XContentType.JSON;
+            String toString = Strings.toString(testInstance);
```

Comment:
same as above, I would use toString here, otherwise we don't test toString but rather toXContent. Maybe this could be a separate test though which verifies that the class is properly marked.